crates/topos-sequencer-subnet-runtime/src/certification.rs

```diff
@@ -52,8 +53,11 @@ impl Certification {
         let subnet_id = self.subnet_id;
         let mut generated_certificates = Vec::new();
 
+        // Keep account of blocks with generated certificates so that we can remove them from
+        // finalized blocks
+        let mut processed_blocks: Vec<u64> = Vec::with_capacity(self.finalized_blocks.len());
```

Comment:
This allocation will be very big at some point right?

Reply:
`finilized_blocks` are only blocks for which certificates are not generated. There could be maybe up to tens of thousands of blocks if TCE network is not available for days so not a big deal

Reply:
Isn't finalized_blocks hardcoded to 256? (`BLOCK_HISTORY_LENGTH`)

Reply:
@dvdplm It was, and then we start dropping blocks if there are more than 256 pending. So this was a bug.